src/shared/components/mutationMapper/MutationMapper.tsx

```diff
@@ -44,12 +46,17 @@ export interface IMutationMapperProps {
 @observer
 export default class MutationMapper<P extends IMutationMapperProps> extends React.Component<P, {}>
 {
+    public static defaultProps: Partial<IMutationMapperProps> = {
+        userSelectionStore: new MutationMapperUserSelectionStore()
```

Comment:
Using the default `MutationMapperUserSelectionStore` value actually shares an implicit singleton instance for all instances of `MutationMapper`. It might be clearer/safer to explicitly create an instance of `MutationMapperSelectionStore` in the parent class, and pass it to the `MutationMapper` instances.

Reply:
I agree this doesnt seem safe.

another option would be to create one in the constructor of the `MutationMapper` instance, and wrap it in a @computed to use the instance version if the prop is undefined

Reply:
I think the best thing would be in contstructor to instantiate a store if one is not passed in props.

Reply:
sounds good, I'll change this to explicitly pass a shared store where we instantiate `MutationMapper`s.

Reply:
@alisman @onursumer is edge case relevant where at one point you pass in a shared store then later on you dont, or vice versa? these types of things can happen at various points as the rendered state stabilizes to the declarative spec